tests/Http/Event/UpdateCalendarRequestHandlerTest.php

```diff
@@ -0,0 +1,728 @@
+                            [new StatusReason(new Language('nl'), 'Covid')]
+                        )
+                    )
+                ),
+            ],
+            'permanent_with_openingHours' => [
+                'data' => (object) [
+                    'calendarType' => 'permanent',
+                    'openingHours' => [
+                        (object) [
+                            'opens' => '10:00',
+                            'closes' => '17:00',
+                            'dayOfWeek' => [
+                                'monday',
+                                'wednesday',
+                            ],
+                        ],
+                        (object) [
+                            'opens' => '8:30',
+                            'closes' => '9:00',
+                            'dayOfWeek' => [
+                                'tuesday',
+                                'thursday',
+                            ],
+                        ],
+                    ],
+                ],
+                'expected_command' => new UpdateCalendar(
+                    self::EVENT_ID,
+                    Calendar::permanent(
+                        [
+                            new OpeningHour(
+                                new OpeningTime(new Hour(10), new Minute(0)),
+                                new OpeningTime(new Hour(17), new Minute(0)),
+                                new DayOfWeekCollection(
+                                    DayOfWeek::MONDAY(),
+                                    DayOfWeek::WEDNESDAY()
+                                )
+                            ),
+                            new OpeningHour(
+                                new OpeningTime(new Hour(8), new Minute(30)),
+                                new OpeningTime(new Hour(9), new Minute(0)),
+                                new DayOfWeekCollection(
+                                    DayOfWeek::TUESDAY(),
+                                    DayOfWeek::THURSDAY()
+                                )
+                            ),
+                        ]
+                    )
+                ),
+            ],
+        ];
+    }
+
+    /**
+     * @test
+     * @dataProvider invalidDataProvider
+     */
+    public function it_throws_an_api_problem_when_given_invalid_data($data, array $expectedSchemaErrors): void
+    {
+        $this->assertCallableThrowsApiProblem(
+            ApiProblem::bodyInvalidData(...$expectedSchemaErrors),
+            fn () => $this->updateCalendarRequestHandler->handle(
+                (new Psr7RequestBuilder())
+                    ->withBodyFromString(json_encode($data))
+                    ->withRouteParameter('eventId', self::EVENT_ID)
+                    ->build('PUT')
+            )
+        );
+
+        $this->assertEquals([], $this->commandBus->getRecordedCommands());
+    }
+
+    public function invalidDataProvider(): array
```

Comment:
We can also add invalid data where:
- The end date is before the start date
- The close hour is before the open hour